include/boost/beast/http/basic_file_body.hpp

```diff
@@ -133,6 +133,19 @@ class basic_file_body<File>::value_type
         return file_size_;
     }
 
+    /// limits the size of the content that can be sent - returns the new size.
+    std::uint64_t
+    limit_size(std::uint64_t limit, error_code & ec)
+    {
+        std::uint64_t actual_size = file_.size(ec);
```

Comment:
By this point in the state of the file object, doesn't the member `file_size_` already have the proper size info? If so why do you need to calculate it again with this automatic variable? If my hypothesis is correct, this method could be a one liner:
`return file_size_ = (std::min)(file_size_, limit);`

Reply:
This is so you can write only a limited amount of the file, i.e. an incomplete chunk

Reply:
To write a range, ok

Reply:
This is weird interface though 'limit size'.. Why not just a `set_range` for both the position and length?

Reply:
Because you might set the limit when you've already read parts of it, i.e. moved the cursor. So you limit the size to 2MB after 1MB is read already. the set_range would reset this.

Reply:
The precondition for `set_range` should be that no reading has occurred yet